### PR TITLE
[CELEBORN-1345] Add a limit to the master's estimated partition size.

### PR DESCRIPTION
**Pull request by RexXiong**
### What changes were proposed in this pull request?
Currently, the Celeborn master calculates the estimatedPartitionSize based on the fileInfo committed by the application. This estimate is then used to allocate slots across all workers. However, this partition size may be too large or too small for Celeborn. For example, if an application commits a single file of 1TB to only one worker, using that partition size could result in all other workers having no available slots or only very small slots. To improve this, it would be better to implement a cap on the master's estimated partition size to prevent such imbalances.

### Why are the changes needed?
As title

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
UT

### CODE REVIEW

**Comment by RexXiong** [comment]
ping @FMX

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 1]
```diff
@@ -2214,10 +2216,19 @@ object CelebornConf extends Logging {
       .bytesConf(ByteUnit.BYTE)
       .createWithDefaultString("64mb")
 
+  val ESTIMATED_PARTITION_SIZE_MAX_SIZE: OptionalConfigEntry[Long] =
+    buildConf("celeborn.master.estimatedPartitionSize.maxSize")
+      .categories("master")
+      .doc(
+        "Max partition size for estimation.")
```
Comment (SteNicholas) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2223]: Adds the description of the default value, which is partitionSplitMaximumSize * 2.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 2]
```diff
@@ -2214,10 +2216,19 @@ object CelebornConf extends Logging {
...
+      .version("0.4.1")
+      .bytesConf(ByteUnit.BYTE)
+      .createOptional
+
   val ESTIMATED_PARTITION_SIZE_MIN_SIZE: ConfigEntry[Long] =
     buildConf("celeborn.master.estimatedPartitionSize.minSize")
       .withAlternative("celeborn.shuffle.minPartitionSizeToEstimate")
-      .categories("worker")
+      .categories("master", "worker")
```
Comment (SteNicholas) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2231]: Adds the reminder of this config in migration.md.

**Review by FMX — APPROVED** [review]
LGTM. Merged into main(v0.5.0) and branch-0.4(v0.4.1).

File: master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/AbstractMetaManager.java [hunk 3]
```diff
@@ -445,7 +445,10 @@ public void updatePartitionSize() {
         Utils.bytesToString(tmpTotalWritten),
         tmpFileCount);
     if (tmpFileCount != 0) {
-      estimatedPartitionSize = tmpTotalWritten / tmpFileCount;
+      estimatedPartitionSize =
+          Math.max(
+              conf.minPartitionSizeToEstimate(),
+              Math.min(tmpTotalWritten / tmpFileCount, conf.maxPartitionSizeToEstimate()));
```
Comment (SteNicholas) [inline master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/AbstractMetaManager.java:451]: Is there a case that estimatedPartitionSize uses `celeborn.master.estimatedPartitionSize.minSize` when `celeborn.master.estimatedPartitionSize.minSize` is greater than `celeborn.master.estimatedPartitionSize.maxSize`?

Comment (RexXiong) [reply master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/AbstractMetaManager.java:451]: > Is there a case that estimatedPartitionSize uses `celeborn.master.estimatedPartitionSize.minSize` when `celeborn.master.estimatedPartitionSize.minSize` is greater than `celeborn.master.estimatedPartitionSize.maxSize`?

That would be `celeborn.master.estimatedPartitionSize.maxSize`, IMO it's not a big problem. :) WDYT?

**Pull request closed without merging**